src/pyro/source.clj

```diff
@@ -23,13 +25,41 @@
   [s n]
   (str "--> " (pad-integer n) " " s))
 
+(defn filepath->lastModified
+  [filepath]
+  (let [url (io/resource filepath)
+        file (io/file filepath)]
+    (cond
+      ; handle files inside JARs
+      ; (io/resource "clojure/java/io.clj") =>
+      ; URL "jar:file:/Users/ivref/.m2/repository/org/clojure/clojure/1.8.0/clojure-1.8.0.jar!/clojure/java/io.clj"
+      (and url (= "jar" (.getProtocol url))) (-> url
```

Comment:
There's enough going on with this threading macro that I'd prefer we move it into its own function.

Reply:
Also, there's no way to clean this up a bit? I haven't done any research here or anything but this seems a bit messy.

Reply:
I agree it's messy. I had an idea though: This is a jar file and it's on the classpath. Is that ever going to change? Could just return 0 when the protocol is `jar`?

Reply:
Seems that this approach will only work for ASCII filenames:
https://stackoverflow.com/questions/20389255/reading-a-resource-file-from-within-jar/20389418#comment59303174_26962028
I will look at it further.